File: tests/gen_ios/test_combined/expected/dd-sdk-ios/Sources/Datadog/Bridge/BridgeWithData.swift
/*
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2016-Present Datadog, Inc.
 */

import Foundation

/**
   An interface to test transferring complex types
 */
@objc(BridgeWithData)
class BridgeWithData: NSObject {

    /**
       Empty method, ComplexDataStructure param, returns void
     */
    func setData(value: ComplexDataStructure) -> Void {
        // TODO
    }

    /**
       Empty method, returns ComplexDataStructure
     */
    func getData() -> ComplexDataStructure {
        // TODO
    }

}

File: tests/gen_android/test_combined/expected/dd-sdk-android/src/main/java/com/datadog/android/bridge/BridgeWithData.kt
/*
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2016-Present Datadog, Inc.
 */

package com.datadog.android.bridge

import android.content.Context

/**
 * An interface to test transferring complex types
 */
class BridgeWithData(context: Context) {

    private val appContext = context.applicationContext

    /**
     * Empty method, ComplexDataStructure param, returns void
     */
    fun setData(value: ComplexDataStructure): Unit {
        TODO()
    }

    /**
     * Empty method, returns ComplexDataStructure
     */
    fun getData(): ComplexDataStructure {
        TODO()
    }

}
